appendix/stata-guide.tex

```diff
@@ -0,0 +1,271 @@
+Relative file paths are common in many other programming languages, but there they are relative to the
+location of the file running the code, and then there is no risk that a file is saved in a completely different folder.
+
+\textbf{Dynamic file paths} use globals that are set in a central master do-file to dynamically build your file
+paths. This have the same function as setting \verb|cd| as all new users should only have to change these
+file path globals in one location. But dynamic absolute file paths are a better practice since if the
+global names are set uniquely there is no risk that files are saved in the incorrect project folder, and
+you can create multiple folder globals instead of just one location as with \verb|cd|.
+
+\codeexample{stata-filepaths.do}{./code/stata-filepaths.do}
+
+\subsection{Line Breaks}
+
+Long lines of code are difficult to read if you have to scroll left and right to see the full line of code. 
+When your line of code is wider than text on a regular paper you should introduce a line break. This breaks 
+the line in the code editor while telling Stata that the same line of code continues on the next row in the 
+code editor. There are two line brakes in Stata and one of them is only intended to be used when doing 
+advanced programming and is discouraged in an article in Stata's official journal.\cite{cox2005styleguide}
+
+Instead, the three forwardslashes should be used. This is a special case of the two forwardslashed comment 
+where the third slash also comments out the new line symbol at the end of that row in the do-file editor.
+
+\codeexample{stata-linebreak.do}{./code/stata-linebreak.do}
+
+\subsection{Boilerplate code}
+
+Boiler plate code is a type of code that always come at the top of the code and its purpose is to harmonize 
+the settings across users running the same code to the biggest degree possible. There is no way in Stata to 
+guarantee that any two installations of Stata will always run code in exactly the same way. In the vast 
+majority of cases it does, but not always, and boilerplate code can mitigate that risk (although not eliminate 
+it). We have developed a command that runs many commonly used boilerplate settings that are optimized given 
+your installation of Stata. It requires two lines to do the setting that mitigates difference in results due 
+to different versions of Stata. 
+
+\codeexample{stata-boilerplate.do}{./code/stata-boilerplate.do}
+
+\subsection{Before Saving}
+
+Similarly to boiler plate code there are good practices that should be followed before saving the data set.	
+These are sorting and ordering the data set, dropping intermediate variables that are not needed and 
+compressing the data set to save disk space and network bandwidth.
+
+If there is an ID variable or a set of ID variables, then the code should also test that they are uniqueally and 
+fully identifying the data set.\sidenote{https://dimewiki.worldbank.org/wiki/ID\_Variable\_Properties} ID 
```

Comment:
@bbdaniels , I cannot get this URL to wrap in the margin, how did you do that?

Reply:
```suggestion
fully identifying the data set.\sidenote{\url{https://dimewiki.worldbank.org/wiki/ID\_Variable\_Properties}} ID 
```

Reply:
Needs the `\url{}` tag -- added in suggestion